pkg/resource/clickpipe.go

```diff
@@ -541,45 +536,63 @@ func (c *ClickPipeResource) Create(ctx context.Context, request resource.CreateR
 
 	if !sourceModel.Kafka.IsNull() {
 		kafkaModel := models.ClickPipeKafkaSourceModel{}
-		credentialsModel := models.ClickPipeKafkaSourceCredentialsModel{}
 		response.Diagnostics.Append(sourceModel.Kafka.As(ctx, &kafkaModel, basetypes.ObjectAsOptions{})...)
-		response.Diagnostics.Append(kafkaModel.Credentials.As(ctx, &credentialsModel, basetypes.ObjectAsOptions{})...)
 
 		var consumerGroup *string
 		if !kafkaModel.ConsumerGroup.IsUnknown() {
 			consumerGroup = kafkaModel.ConsumerGroup.ValueStringPointer()
 		}
 
-		credentials := &api.ClickPipeKafkaSourceCredentials{}
-
-		if !credentialsModel.Username.IsNull() && !credentialsModel.Password.IsNull() {
-			credentials.ClickPipeSourceCredentials = &api.ClickPipeSourceCredentials{
-				Username: credentialsModel.Username.ValueString(),
-				Password: credentialsModel.Password.ValueString(),
-			}
-		} else if !credentialsModel.AccessKeyID.IsNull() && !credentialsModel.SecretKey.IsNull() {
-			credentials.ClickPipeSourceAccessKey = &api.ClickPipeSourceAccessKey{
-				AccessKeyID: credentialsModel.AccessKeyID.ValueString(),
-				SecretKey:   credentialsModel.SecretKey.ValueString(),
-			}
-		} else if !credentialsModel.ConnectionString.IsNull() {
-			credentials.ConnectionString = credentialsModel.ConnectionString.ValueStringPointer()
-		} else {
-			response.Diagnostics.AddError(
-				"Error Creating ClickPipe",
-				"Kafka source requires credentials",
-			)
-			return
-		}
-
 		clickPipe.Source.Kafka = &api.ClickPipeKafkaSource{
 			Type:           kafkaModel.Type.ValueString(),
 			Format:         kafkaModel.Format.ValueString(),
 			Brokers:        kafkaModel.Brokers.ValueString(),
 			Topics:         kafkaModel.Topics.ValueString(),
 			ConsumerGroup:  consumerGroup,
 			Authentication: kafkaModel.Authentication.ValueString(),
-			Credentials:    credentials,
+			IAMRole:        kafkaModel.IAMRole.ValueStringPointer(),
+			CACertificate:  kafkaModel.CACertificate.ValueStringPointer(),
+		}
+
+		if kafkaModel.Authentication.ValueString() != api.ClickPipeAuthenticationIAMRole {
+			if !kafkaModel.Credentials.IsNull() {
+				credentialsModel := models.ClickPipeKafkaSourceCredentialsModel{}
+				response.Diagnostics.Append(kafkaModel.Credentials.As(ctx, &credentialsModel, basetypes.ObjectAsOptions{})...)
+
+				var credentials *api.ClickPipeKafkaSourceCredentials
+
+				if kafkaModel.Authentication.ValueString() != api.ClickPipeAuthenticationIAMRole {
```

Comment:
This is the same we're checking a few lines above. Can probably get rid of the check as it will always be true.